openfe/protocols/openmm_rfe/equil_rfe_methods.py

```diff
@@ -587,107 +591,110 @@ def run(self, *, dry=False, verbose=True,
 
-        sampler.energy_context_cache = energy_context_cache
-        sampler.sampler_context_cache = sampler_context_cache
+            if not dry:  # pragma: no-cover
+                # minimize
+                if verbose:
+                    logger.info("minimizing systems")
 
-        if not dry:  # pragma: no-cover
-            # minimize
-            if verbose:
-                logger.info("minimizing systems")
+                sampler.minimize(max_iterations=sim_settings.minimization_steps)
 
-            sampler.minimize(max_iterations=sim_settings.minimization_steps)
+                # equilibrate
+                if verbose:
+                    logger.info("equilibrating systems")
 
-            # equilibrate
-            if verbose:
-                logger.info("equilibrating systems")
+                sampler.equilibrate(int(equil_steps.m / mc_steps))  # type: ignore
 
-            sampler.equilibrate(int(equil_steps.m / mc_steps))  # type: ignore
+                # production
+                if verbose:
+                    logger.info("running production phase")
 
-            # production
-            if verbose:
-                logger.info("running production phase")
+                sampler.extend(int(prod_steps.m / mc_steps))  # type: ignore
 
-            sampler.extend(int(prod_steps.m / mc_steps))  # type: ignore
+                # calculate estimate of results from this individual unit
+                ana = multistate.MultiStateSamplerAnalyzer(reporter)
+                est, _ = ana.get_free_energy()
+                est = (est[0, -1] * ana.kT).in_units_of(omm_unit.kilocalories_per_mole)
+                est = ensure_quantity(est, 'openff')
 
-            # calculate estimate of results from this individual unit
-            ana = multistate.MultiStateSamplerAnalyzer(reporter)
-            est, _ = ana.get_free_energy()
-            est = (est[0, -1] * ana.kT).in_units_of(omm_unit.kilocalories_per_mole)
-            est = ensure_quantity(est, 'openff')
+                # close reporter when you're done
+                reporter.close()
 
-            # close reporter when you're done
-            reporter.close()
+                nc = shared_basepath / sim_settings.output_filename
+                chk = shared_basepath / sim_settings.checkpoint_storage
+            else:
+                # close reporter when you're done, prevent file handle clashes
+                reporter.close()
 
-            nc = shared_basepath / sim_settings.output_filename
-            chk = shared_basepath / sim_settings.checkpoint_storage
+                # clean up the reporter file
+                fns = [shared_basepath / sim_settings.output_filename,
+                       shared_basepath / sim_settings.checkpoint_storage]
+                for fn in fns:
+                    os.remove(fn)
+        finally:
+            energy_context_cache.empty()
+            sampler_context_cache.empty()
```

Comment:
this feels like it should get (eventually) upstreamed as these context objects are used using a context manager pattern

Reply:
I think in the short term solution, it might be good to at least make sure that the context is wiped in multistatesampler.__del__

Reply:
Indeed, upstreaming here: https://github.com/choderalab/openmmtools/pull/690

Reply:
Also cc https://github.com/choderalab/openmmtools/issues/688